docs/Tutorials/CheckpointRestart.md

```diff
@@ -53,3 +51,39 @@ and restarting:
    (that slow the code) and too-infrequent synchronizations (that won't allow
    checkpoints to be written).
 
+Certain simulation parameters can be modified when restarting from a checkpoint
+file. This is done by parsing a new input file containing just those options to
+modify; all other options will preserve their value from the original run.
+
+Note, however, that not all tags are permitted to be modified: in the current
+implementation, only tags from the `const_global_cache_tags` that also have a
+member variable `static constexpr bool is_overlayable = true;` can be modified.
+The reason for this "opt-in" design is that in general, most tags interact with
+past or current simulation data in a way that would invalidate the simulation
+state if the tag were modified on restart (example: changing the domain
+invalidates all spatial data, changing the timestepper invalidates the history).
+Only tags that do not interact with the state should be permitted to be updated.
+For example: activation thresholds on various algorithms, or frequency of data
+observation, are safe parameters to modify.
+
+The executable will update the global cache with new input file values during
+the phase `UpdateOptionsAtRestartFromCheckpoint`. The
+`CheckpointAndExitAfterWallclock` phase control automatically directs code flow
+to this phase after a restart.
+
+TODO FOR CODE REVIEW: If someone uses a
+`VisitAndReturn(UpdateOptionsAtRestartFromCheckpoint)` just after a
+`VisitAndReturn(WriteCheckpoint)`, the code will try to parse the overlay files
+during evolution, even if no restarts occur. In other words, the code might
+write a checkpoint every 100 slabs, and will try to reparse the input file after
+each one. Since it's hard to know if there was a restart in your past (except
+perhaps by making guesses from the wallclock time?), it's hard to handle this
+sanely...
```

Comment:
If Main cannot migrate (or if you can distinguish a migration from a checkpoint), you could keep track of whether a restart has completed using a non-checkpointed member variable.